add_gist.rb

```diff
@@ -0,0 +1,66 @@
+require 'rubygems'
+require 'net/http'
+require 'uri'
+require 'json'
+require 'OpenSSL'
+require 'find'
+require 'dotenv/load'
+
+module AddGist
+  GISTS_URL = 'https://api.github.com/gists'.freeze
+
+  def self.upload_files(path, options = {})
+    files = read_files(path)
+    return nil if files.empty?
+
+    response = send_request(files, options)
+
+    if response.is_a?(Net::HTTPCreated)
+      html_url = JSON.parse(response.body)['html_url']
+      puts "Gist created successfully! \nAccess URL: #{html_url}"
+    else
+      puts response.code
+    end
+  end
+
+  def self.read_files(pwd = Dir.pwd)
+    files = {}
+    Find.find(pwd) do |path|
+      files.merge!(File.basename(path) => { 'content': File.open(path, 'r+').read }) unless File.directory?(path)
+    end
+    files
+  rescue Errno::ENOENT => e
+    puts e.message
+  end
+
+  def self.send_request(files, options = {})
+    uri = URI.parse(GISTS_URL)
+    uri.query = URI.encode_www_form(access_token: ENV['ACCESS_TOKEN'])
+
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+    request = Net::HTTP::Post.new(uri.request_uri)
+    request.body = {
+      description: options[:description].to_s,
+      public: options[:is_public],
+      files: files
+    }.to_json
+
+    http.request(request)
+  end
+
+  private_class_method :read_files, :send_request
```

Comment:
can you check if this sentence goes here best instead of at the beginning of the module

Reply:
If I change it to the beginning of the module, a ```NameError``` comes up. 

I found this: 
> Notice that the call to private_class_method has to come after the method is defined. It will not work if it is passed the method name that is not yet defined.
http://tmr08c.github.io/classes/2015/04/09/private-class-methods.html